### PR TITLE
Set CONDA_OVERRIDE_GLIBC

### PR DESCRIPTION
**Pull request by isuruf**
Fixes https://github.com/regro/cf-scripts/issues/1207

### CODE REVIEW

**Comment by jakirkham** [comment]
Thanks Isuru! 😄

**Review by beckermr — APPROVED** [review]

**Comment by beckermr** [comment]
@isuruf how did the virtual package get in the list in the first place? Given that mamba is doing the checks here, is it going to respect the conda overrides?

**Comment by jakirkham** [comment]
> Given that mamba is doing the checks here, is it going to respect the conda overrides?

Yep it does. See issue ( https://github.com/mamba-org/mamba/issues/84 ) for more context.

File: conda_forge_tick/mamba_solver.py [hunk 1]
```diff
@@ -379,5 +382,8 @@ def _is_recipe_solvable_on_platform(recipe_dir, cbc_path, platform, arch):
             solvable = solvable and _solvable
             if _err is not None:
                 errors.append(_err)
-
+
```
Comment (beckermr) [inline conda_forge_tick/mamba_solver.py:385]:
```suggestion
```

File: conda_forge_tick/mamba_solver.py [hunk 2]
```diff
@@ -290,6 +290,9 @@ def _clean_reqs(reqs, names):
 
 
 def _is_recipe_solvable_on_platform(recipe_dir, cbc_path, platform, arch):
+    # Make sure mamba doesn't choke on virtual package solving
+    if platform == "linux":
+        os.environ["CONDA_OVERRIDE_GLIBC"] = "2.31"
```
Comment (beckermr) [inline conda_forge_tick/mamba_solver.py:295]:
```suggestion
        os.environ["CONDA_OVERRIDE_GLIBC"] = "2.31"

```

File: conda_forge_tick/mamba_solver.py [hunk 3]
```diff
@@ -379,5 +383,7 @@ def _is_recipe_solvable_on_platform(recipe_dir, cbc_path, platform, arch):
             solvable = solvable and _solvable
             if _err is not None:
                 errors.append(_err)
-
+    if platform == "linux":
+        del os.environ["CONDA_OVERRIDE_GLIBC"]
+
```
Comment (beckermr) [inline conda_forge_tick/mamba_solver.py:388]:
```suggestion

```

**Pull request merged**